### DISCUSSION TITLE
How to document vlan across sites when using netbox-sync?

### DISCUSSION THREAD

**Discussion by sliddjur** [bb-Ricardo/netbox-sync#180] [Q&A] [answered]
Hello and thanks for the great work you put into this tool! 

I am wondering how I should go on to document a few things together with netbox-sync

We have two geographically separated sites, but only one vcenter. The vlans are stretched with help of vxlan, so from a networking perspective in vcenter, its considered one site. 

Now, netbox-sync will want to bind a cluster to a site, and put a host in that cluster, and vms on the host, and portgroups/vlans on the vms. So eventually we end up with duplicate vlans (portgroups) on two different sites if we do some filtering on the physical sites.

What I tried to do is deleting the clusters, and moving netbox-sync synced hosts to the physical site. But that ended up not being able to sync vms, ips and other objet on those hosts anymore...

And we have already documented the physical part of the two sites. So it seems to just end up in a mess when trying to play with netbox-sync and multiple sites.

Is a better approach to use one site in netbox, and do the physical separation based on netbox "Locations"?
I want to document the hosts into my physical sites in the appropriate racks. Also some other physical servers are using the same vlan (ids).

**Comment by bb-Ricardo** [comment]
Hi,

I understand the issue. I guess I have to setup an test environment where I have the same VLAN in two sides.

I mean currently the easiest sulution would be to define thw VLAN or subnet as global and not site specific. If this subnet range exists only once in your infrastructure then this could be an easy solution.

But also the vxlan scenario should somehow be supported.

**Reply by sliddjur** [reply]
Subnets and vlans both exist only once in my infrastructure.
But the vlans (portgroup/nics) created are bound to a site. 
Can netbox-sync make cluster/hosts/vms/nics site-independant?

**Comment by sliddjur** [comment, marked as answer]
I have solved this with pre-creating all VLANs in netbox, and *not* assigning it to a site. 
Luckily, this script will not recreate the vlan when it exists the same vlan-id globally.
This is the behavior I wanted, so all good.

If netbox-sync syncs a vlan that I havent defined, I can just delete the site-relation in the vlan-object and go on from there!

**Reply by bb-Ricardo** [reply]
Hi, that's exactly what I tried to say. Probably not well enough 😬. If a VLAN with the same ID exists globally and NOT in the site then the global VLAN is used.
